### PR TITLE
Deploy

### PR DESCRIPTION
**Pull request by andersonresende**
- Vercell deploy
- Missing add to Readme: (In progress)
-  Add button to frontend worked pretty well
- Add button to backend is failing

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -63,4 +63,11 @@ repos:
         language: system
         # Only run frontend client generation if frontend files have changed:
         files: openapi\.json$
+        pass_filenames: false
+        # export requirements is required to vercel deploy
```
Comment (andersonresende) [inline .pre-commit-config.yaml:67]: Vercell deploy requires a requirements.txt file.

File: fastapi_backend/api/index.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from app.main import app  # noqa: F401
```
Comment (andersonresende) [inline fastapi_backend/api/index.py:1]: #noqa because ruff was complaining there's an unused import.

Comment (andersonresende) [reply fastapi_backend/api/index.py:1]: This is the file from which vercell reads and creates the serveless functions.
It needs awsgi application: `app`

File: fastapi_backend/vercel.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (andersonresende) [inline fastapi_backend/vercel.json:1]: We need to point the file that has the routes.

File: fastapi_backend/alembic.ini [hunk 4]
```diff
@@ -3,7 +3,7 @@
 [alembic]
 # path to migration scripts.
 # Use forward slashes (/) also on windows to provide an os agnostic path
-script_location = alembic
+script_location = alembic_migrations
```
[the remark this answers is not in the document]
Comment (fjsj) [reply fastapi_backend/alembic.ini:6]: Good catch. In general we should avoid any internal modules with the same name of third-party libs.

File: fastapi_backend/app/config.py [hunk 5]
```diff
@@ -30,6 +33,9 @@ class Settings(BaseSettings):
     # Frontend
     FRONTEND_URL: str = "http://localhost:3000"
 
+    # CORS
+    CORS_ORIGINS: Set[str] = Field(default_factory=lambda: set("*"))
```
[the remark this answers is not in the document]
Comment (fjsj) [reply fastapi_backend/app/config.py:37]: It should be part of prod env vars. Also update README about this.

File: fastapi_backend/app/database.py [hunk 6]
```diff
@@ -2,12 +2,13 @@
...
 from fastapi import Depends
 from fastapi_users.db import SQLAlchemyUserDatabase
+from sqlalchemy import NullPool
 from sqlalchemy.ext.asyncio import AsyncSession, async_sessionmaker, create_async_engine
 
 from .config import settings
 from .models import Base, User
 
-engine = create_async_engine(settings.DATABASE_URL)
+engine = create_async_engine(settings.DATABASE_URL, poolclass=NullPool)
```
[the remark this answers is not in the document]
Comment (fjsj) [reply fastapi_backend/app/database.py:11]: Just add a comment here saying disabling connection pool is necessary for Vercel / serverless.

**Review by fjsj — CHANGES_REQUESTED** [review]
Overall LGTM, check comments.

File: fastapi_backend/app/config.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (andersonresende) [reply fastapi_backend/app/config.py]: Removed the default. I had already included that to the .env.example

I'll adjust the Readme into the deployment section, next PR.

**Review by fjsj — APPROVED** [review]

**Pull request merged**